nselib/url.lua

```diff
@@ -173,12 +177,31 @@ function parse(url, default)
     return ""
   end)
   -- get params
-  url = string.gsub(url, "%;(.*)", function(p)
-    parsed.params = p
-    return ""
-  end)
+  -- Split up the query, if necessary
+  if(parsed.query) then
+    parsed.params = {}
```

Comment:
As discussed above, this key should not be "params" because that has a different meaning. I suggest "querystring" since that is what http.lua used for this table of key-value pairs. In fact, for each of the old key names (under http.parse_url) you should check through the code looking for their use and change it if necessary.